java/src/test/java/vowpalWabbit/learner/VWActionProbsLearnerTest.java

```diff
@@ -0,0 +1,136 @@
+package vowpalWabbit.learner;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import vowpalWabbit.VWTestHelper;
+import vowpalWabbit.responses.ActionProb;
+import vowpalWabbit.responses.ActionProbs;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertArrayEquals;
+
+/**
+ * Created by jmorra on 10/2/15.
+ */
+public class VWActionProbsLearnerTest extends VWTestHelper {
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Test
+    public void testCBExplore() throws IOException {
+        String model = temporaryFolder.newFile().getAbsolutePath();
+        String[] cbTrain = new String[]{
+            "1:2:0.4 | a c",
+            "3:0.5:0.2 | b d",
+            "4:1.2:0.5 | a b c",
+            "2:1:0.3 | b c",
+            "3:1.5:0.7 | a d"
+        };
+
+        VWActionProbsLearner vw = VWLearners.create("--quiet --cb_explore 4 -f " + model);
+        ActionProbs[] trainPreds = new ActionProbs[cbTrain.length];
+        for (int i=0; i<cbTrain.length; ++i) {
+            trainPreds[i] = vw.learn(cbTrain[i]);
+        }
+        ActionProbs[] expectedTrainPreds = new ActionProbs[]{
```

Comment:
Stupid thought, but do you think this would be more readable as just:

```java
ActionProbs[] expectedTrainPreds = new ActionProbs[]{
  actionProbs(0.9625f, 0.0125f, 0.0125f, 0.0125f),
  actionProbs(0.0125f, 0.9625f, 0.0125f, 0.0125f),
  ...
};
```

Then you could have a function, `actionProbs` that takes a `float ...` *varargs* and returns a `ActionProbs` by zipping the probability with the index in the *varargs* array.  This seems like it would be more readable and more terse.